api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto

```diff
@@ -111,6 +112,14 @@ message RateLimit {
   // in case of rate limiting (i.e. 429 responses).
   // Having this header not present potentially makes the request retriable.
   bool disable_x_envoy_ratelimited_header = 9;
+
+  // This field allows for a custom HTTP response status code to the downstream client when
+  // the request has been rate limited.
+  // Defaults to 429 (TooManyRequests).
+  //
+  // .. note::
+  //   If this is set to < 400, 429 will be used instead.
+  type.v3.HttpStatus status = 10;
```

Comment:
Does the existing support in https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#envoy-v3-api-msg-extensions-filters-network-http-connection-manager-v3-localreplyconfig not work here?

Reply:
it works,but there are still some problems :
1、it seems that the statistics emitted by rate limit filter is still 429
2、it works on the http connection manager,it seems that the scope of influence is a little large,we just want config response code for some specific rate limit,and if i have two rate limiter want to return different code，it seems not works（such as two rate_limit but different stage or local_rate_limit and rate_limit）
consider envoy local_rate_limit already has status field to custom response code，i think it's fine to add this field for rate_limit

Reply:
OK, seems reasonable.